lib/services/project-data-service.ts

```diff
@@ -51,6 +53,137 @@ export class ProjectDataService implements IProjectDataService {
+		const splashImages = await this.getIOSAssetSubGroup(pathToSplashImages);
+
+		return {
+			icons,
+			splashBackgrounds,
+			splashCenterImages,
+			splashImages
+		};
+	}
+
+	@exported("projectDataService")
+	public async getAndroidAssetsStructure(opts: IProjectDir): Promise<IAssetGroup> {
+		// TODO: Use image-size package to get the width and height of an image.
+		// TODO: Parse the splash_screen.xml in nodpi directory and get from it the names of the background and center image.
+		// TODO: Parse the AndroidManifest.xml to get the name of the icon.
+		// This way we'll not use the image-definitions.json and the method will return the real android structure.
+		const projectDir = opts.projectDir;
+		const projectData = this.getProjectData(projectDir);
+		const pathToAndroidDir = path.join(projectData.appResourcesDirectoryPath, this.$devicePlatformsConstants.Android);
+		const hasMigrated = this.$androidResourcesMigrationService.hasMigrated(projectData.appResourcesDirectoryPath);
+		const basePath = hasMigrated ? path.join(pathToAndroidDir, SRC_DIR, MAIN_DIR, RESOURCES_DIR) : pathToAndroidDir;
+
+		const currentStructure = this.$fs.enumerateFilesInDirectorySync(basePath);
+		const content = this.getImageDefinitions().android;
+
+		return {
+			icons: this.getAndroidAssetSubGroup(content.icons, currentStructure),
+			splashBackgrounds: this.getAndroidAssetSubGroup(content.splashBackgrounds, currentStructure),
+			splashCenterImages: this.getAndroidAssetSubGroup(content.splashCenterImages, currentStructure),
+			splashImages: null
+		};
+	}
+
+	private getImageDefinitions(): any {
+		const pathToImageDefinitions = path.join(__dirname, "..", "..", CLI_RESOURCES_DIR_NAME, AssetConstants.assets, AssetConstants.imageDefinitionsFileName);
+		const imageDefinitions = this.$fs.readJson(pathToImageDefinitions);
+
+		return imageDefinitions;
+	}
+
+	private async getIOSAssetSubGroup(dirPath: string): Promise<IAssetSubGroup> {
+		const pathToContentJson = path.join(dirPath, AssetConstants.iOSResourcesFileName);
+		const content = <IAssetSubGroup>this.$fs.readJson(pathToContentJson);
+
+		const imageDefinitions = this.getImageDefinitions().ios;
+
+		if (content && content.images) {
+			_.each(content.images, image => {
+				// In some cases the image may not be available, it will just be described.
+				// When this happens, the filename will be empty.
+				// So we'll keep the path empty as well.
+				if (image.filename) {
+					image.path = path.join(dirPath, image.filename);
+				}
+
+				if (image.size) {
+					// size is basically <width>x<height>
+					const [width, height] = image.size.toString().split(AssetConstants.sizeDelimiter);
+					if (width && height) {
+						image.width = +width;
+						image.height = +height;
+					}
+				} else {
+					// Find the image size based on the hardcoded values in the image-definitions.json
+					_.each(imageDefinitions, (assetSubGroup: IAssetItem[]) => {
```

Comment:
You shouldn't need to point the type of `assetSubGroup` out if `imageDefinitions` was strongly typed - I believe its type is actually `IAssetsStructure`